src/Symfony/Component/HttpKernel/EventListener/HttpExceptionListener.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\EventListener;
+
+use Symfony\Component\Debug\Exception\FlattenException;
+use Symfony\Component\Debug\ExceptionHandler;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Event\GetResponseForExceptionEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/**
+ * @author Chris McCafferty <cilefen@gmail.com>
+ */
+class HttpExceptionListener implements EventSubscriberInterface
+{
+    protected $debug;
+    protected $charset;
+
+    public function __construct($debug = false, $charset = null)
+    {
+        $this->debug = $debug;
+        $this->charset = $charset;
+    }
+
+    public static function getSubscribedEvents()
+    {
+        return array(
+            KernelEvents::EXCEPTION => array(
+                array('onKernelException', -256),
+            ),
+        );
+    }
+
+    public function onKernelException(GetResponseForExceptionEvent $event)
+    {
+        if ($event->hasResponse()) {
+            return;
+        }
+
+        $flattenException = FlattenException::create($event->getException());
+        $handler = new ExceptionHandler($this->debug);
+        $headers = $flattenException->getHeaders();
+        header(sprintf('HTTP/1.0 %s', $flattenException->getStatusCode()));
+        $headers['Content-Type'] = 'text/html; charset='.$this->charset;
```

Comment:
you should be able to remove the 3 lines above (and inline $flattenException->getHeaders() below) I suppose
sending the actual headers will be handled by HttpKernel